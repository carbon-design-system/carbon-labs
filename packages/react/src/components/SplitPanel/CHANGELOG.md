# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.4.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-split-panel@0.3.0...@carbon-labs/react-split-panel@0.4.0) (2025-05-28)


### Bug Fixes

* update exports to match UI shell ([#605](https://github.com/carbon-design-system/carbon-labs/issues/605)) ([1bf4682](https://github.com/carbon-design-system/carbon-labs/commit/1bf46822620b44cc1ad9ce58913bf26e9c3a2ca1))





# [0.3.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-split-panel@0.2.0...@carbon-labs/react-split-panel@0.3.0) (2025-04-25)

**Note:** Version bump only for package @carbon-labs/react-split-panel





# 0.2.0 (2025-04-22)


### Features

* create split panel component ([#501](https://github.com/carbon-design-system/carbon-labs/issues/501)) ([a579b97](https://github.com/carbon-design-system/carbon-labs/commit/a579b978a085b38f4d2f7ea4b853a29e022632b0))





# 0.1.0 (2025-03-06)


### Features

* script to generate base component in react + web-components ([#354](https://github.com/carbon-design-system/carbon-labs/issues/354)) ([3c24d37](https://github.com/carbon-design-system/carbon-labs/commit/3c24d375f2b8773f7c4dd704197ae6267d919685))
