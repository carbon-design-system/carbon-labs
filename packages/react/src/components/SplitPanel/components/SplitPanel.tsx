/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { useRef, useState } from 'react';
import { usePrefix } from '@carbon-labs/utilities/es/index.js';
import cx from 'classnames';
import { DragHorizontal, DragVertical } from '@carbon/react/icons';

/** Primary UI component for user interaction */

interface SplitPanelProps {
  /**
   * children placed in the panel after the split
   */
  childrenAfterSplit: React.ReactNode;
  /**
   * children placed in the panel before the split
   */
  childrenBeforeSplit: React.ReactNode;
  /**
   * class to add to the outer component
   */
  className?: string;
  /**
   * initial position/value (0 to 100) of splitter, defaults to 50.
   */
  defaultSplitValue?: number;
  /**
   * gap between panels used to handle the split
   * sizes default, narrow and none correspond to Carbon grid default,
   * narrow and condensed gaps
   */
  gap?: 'default' | 'narrow' | 'none';
  /**
   * onChange event reporting the current split value
   * @param splitValue
   * @returns
   */
  onChange?: (splitValue: number) => void;
  /**
   *  panel arrangement is horizontal or vertical
   */
  orientation?: 'horizontal' | 'vertical';
  /**
   *  lowest possible split value and default 0
   */
  splitMin?: number;
  /**
   *  highest possible split value and default 100
   */
  splitMax?: number;
  /**
   *  Number of pixels moved using cursor keys
   */
  splitKeyStepPixels?: number;
  /**
   *  Number of pixels moved using shift and  cursor keys
   */
  splitShiftKeyStepPixels?: number;
}

type handledKeyType = {
  ArrowUp: boolean;
  ArrowDown: boolean;
  ArrowLeft: boolean;
  ArrowRight: boolean;
  Home: boolean;
  End: boolean;
};

const handledKeys = [
  'ArrowUp',
  'ArrowDown',
  'ArrowLeft',
  'ArrowRight',
  'Home',
  'End',
];

const toDP = (num, dp) => {
  return parseFloat(num.toFixed(dp));
};

const sanitizeValue = (value, min = 0, max = 100) =>
  Math.min(max, Math.max(min, value));

export const SplitPanel = ({
  childrenBeforeSplit,
  childrenAfterSplit,
  className,
  defaultSplitValue = 50,
  gap = 'default',
  onChange,
  orientation = 'horizontal',
  splitMin = 0,
  splitMax = 100,
  splitKeyStepPixels = 5,
  splitShiftKeyStepPixels = 25,
  ...rest
}: SplitPanelProps) => {
  const prefix = usePrefix();
  const splitPanelRef = useRef<HTMLInputElement>(null);
  const splitHandleRef = useRef<HTMLInputElement>(null);
  const mousePrev = useRef<{ value: number } | null>(null);
  const [kbInteraction, setKbInteraction] = useState(false);
  const panelSize = useRef(1);
  const isHorizontal = orientation === 'horizontal';
  const localSplitMin = sanitizeValue(splitMin);
  const localSplitMax = sanitizeValue(splitMax);
  // keep value between localSplitMin and localSplitMax
  const [splitValue, setSplitValue] = useState<number>(
    sanitizeValue(defaultSplitValue, localSplitMin, localSplitMax)
  );

  const updatePanelSize = () => {
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const splitPanelRect = splitPanelRef.current!.getBoundingClientRect();
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const splitRect = splitHandleRef.current!.getBoundingClientRect();

    panelSize.current = isHorizontal
      ? splitPanelRect.width - splitRect.width
      : splitPanelRect.height - splitRect.height;
  };

  const splitMouseDown = (ev) => {
    setKbInteraction(false);
    updatePanelSize();
    mousePrev.current = { value: isHorizontal ? ev.clientX : ev.clientY };
  };

  const splitMouseUp = () => {
    mousePrev.current = null;
  };

  const doSplitChange = (change) => {
    setSplitValue((prevSplitValue) => {
      const splitStart = (prevSplitValue * panelSize.current) / 100;

      const newPosition = sanitizeValue(
        (100 * (splitStart + change)) / panelSize.current,
        localSplitMin,
        localSplitMax
      );

      const result = toDP(newPosition, 3);
      onChange?.(result);
      return result;
    });
  };

  const splitMouseMove = (ev) => {
    if (!mousePrev?.current) {
      return;
    }

    const change =
      (isHorizontal ? ev.clientX : ev.clientY) - mousePrev.current.value;

    // update previous mouse position
    mousePrev.current = { value: isHorizontal ? ev.clientX : ev.clientY };

    doSplitChange(change);
  };

  const style = {
    [`--${prefix}--split-panel-before-size`]: `${splitValue / 100}fr`,
    [`--${prefix}--split-panel-after-size`]: `${1 - splitValue / 100}fr`,
  } as React.CSSProperties;

  const splitKeyMove = (ev) => {
    if (!handledKeys.includes(ev.key)) {
      return;
    }

    const key = { [`${ev.key}`]: true } as handledKeyType;

    if (
      (isHorizontal && (key.ArrowUp || key.ArrowDown)) ||
      (!isHorizontal && (key.ArrowLeft || key.ArrowRight))
    ) {
      return;
    }

    updatePanelSize();
    setKbInteraction(true);

    if (key.Home) {
      setSplitValue(localSplitMin);
    } else if (key.End) {
      setSplitValue(localSplitMax);
    } else {
      let change;
      if ((isHorizontal && key.ArrowLeft) || (!isHorizontal && key.ArrowUp)) {
        change =
          -1 * (ev.shiftKey ? splitShiftKeyStepPixels : splitKeyStepPixels);
      }
      if (
        (isHorizontal && key.ArrowRight) ||
        (!isHorizontal && key.ArrowDown)
      ) {
        change = ev.shiftKey ? splitShiftKeyStepPixels : splitKeyStepPixels;
      }

      doSplitChange(change);
    }
  };

  return (
    <div
      className={cx(className, `${prefix}--split-panel`)}
      {...rest}
      ref={splitPanelRef}>
      {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
      <div
        className={cx(
          `${prefix}--split-panel__content--${orientation}`,
          `${prefix}--split-panel__content--gap-${gap}`,
          {
            [`${prefix}--split-panel__content--kb-interaction`]: kbInteraction,
          }
        )}
        // eslint-disable-next-line react/forbid-dom-props
        style={style}
        onMouseMove={splitMouseMove}
        onMouseUp={splitMouseUp}
        onMouseLeave={splitMouseUp}>
        <div className={`${prefix}--split-panel__panel`}>
          {childrenBeforeSplit}
        </div>
        {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
        <div
          className={`${prefix}--split-panel__handle--${orientation}`}
          role="separator"
          aria-orientation={orientation}
          // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
          tabIndex={0}
          aria-valuenow={splitValue}
          aria-valuemin={localSplitMin}
          aria-valuemax={localSplitMax}
          data-key-step={splitKeyStepPixels}
          data-key-shift-step={splitShiftKeyStepPixels}
          ref={splitHandleRef}
          onMouseDown={splitMouseDown}
          onKeyDown={splitKeyMove}>
          {orientation === 'horizontal' ? (
            <DragHorizontal
              className={`${prefix}--split-panel__handle-icon`}
              size={20}
            />
          ) : (
            <DragVertical
              className={`${prefix}--split-panel__handle-icon`}
              size={20}
            />
          )}
        </div>
        <div className={`${prefix}--split-panel__panel`}>
          {childrenAfterSplit}
        </div>
      </div>
    </div>
  );
};
