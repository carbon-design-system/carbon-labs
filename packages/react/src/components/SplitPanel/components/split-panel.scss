/**
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
@use '@carbon/styles/scss/theme' as *;
@use '@carbon/styles/scss/colors' as *;
@use '@carbon/styles/scss/spacing' as *;
@use '@carbon/styles/scss/motion' as *;

$prefix: 'clabs' !default;

.#{$prefix}--split-panel {
  overflow: hidden;
  block-size: 100%;
}

[class*='#{$prefix}--split-panel__content--'] {
  --#{$prefix}--split-panel-before-size: 0.5fr;
  --#{$prefix}--split-panel-after-size: 0.5fr;
  --#{$prefix}--split-panel-gap: #{$spacing-07};
  --#{$prefix}--split-panel-handle-transition: initial;

  display: grid;
  overflow: hidden;
  block-size: 100%;

  transition: var(--#{$prefix}--split-panel-handle-transition);
}

.#{$prefix}--split-panel__content--gap-narrow {
  --#{$prefix}--split-panel-gap: #{$spacing-05};
}

.#{$prefix}--split-panel__content--gap-none {
  --#{$prefix}--split-panel-gap: 0;
}

.#{$prefix}--split-panel__content--horizontal {
  grid-template-columns:
    var(--#{$prefix}--split-panel-before-size) var(
      --#{$prefix}--split-panel-gap
    )
    var(--#{$prefix}--split-panel-after-size);
}

.#{$prefix}--split-panel__content--vertical {
  grid-template-rows:
    var(--#{$prefix}--split-panel-before-size) var(
      --#{$prefix}--split-panel-gap
    )
    var(--#{$prefix}--split-panel-after-size);
}

.#{$prefix}--split-panel__content--kb-interaction {
  /* linear looks smoother to my eye than motion(standard, productive) */
  --#{$prefix}--split-panel-handle-transition: all #{$duration-moderate-01} linear;
}

.#{$prefix}--split-panel__panel {
  overflow: auto;
  /* padding causes panel sizing issue when larger than space available */
  padding: 0;
  background-color: $layer;
  block-size: 100%;
}

/* panels that directly contain another split panel do not have a background color so that the
split handle remains transparent through to the background color */
.#{$prefix}--split-panel__panel:has(
    > .#{$prefix}--split-panel .#{$prefix}--split-panel__panel
  ) {
  background-color: transparent;
}

[class*='#{$prefix}--split-panel__handle--'] {
  --#{$prefix}--split-panel-handle-color: #{$border-subtle-01};
  --#{$prefix}--split-panel-handle-border-color: transparent;
  --#{$prefix}--split-panel-handle-size: 1px;
  --#{$prefix}--split-panel-handle-inline-size: var(
    --#{$prefix}--split-panel-handle-size
  );
  --#{$prefix}--split-panel-handle-block-size: var(
    --#{$prefix}--split-panel-handle-size
  );
  --#{$prefix}--split-panel-handle-cursor: initial;

  position: relative;
  cursor: var(--#{$prefix}--split-panel-handle-cursor);
}

[class*='#{$prefix}--split-panel__handle--']::after {
  position: absolute;
  display: block;
  box-sizing: content-box;
  border: 1px solid var(--#{$prefix}--split-panel-handle-border-color);
  background-clip: content-box;
  background-color: var(--#{$prefix}--split-panel-handle-color);
  block-size: var(--#{$prefix}--split-panel-handle-block-size);
  content: '';
  inline-size: var(--#{$prefix}--split-panel-handle-inline-size);
  inset-block-start: 50%;
  inset-inline-start: 50%;
  transform: translate(-50%, -50%);
  transition: all $fast-02 $ease-in;
}

[class*='#{$prefix}--split-panel__handle--']:hover {
  --#{$prefix}--split-panel-handle-color: #{$border-strong-01};
}

[class*='#{$prefix}--split-panel__handle--']:focus {
  --#{$prefix}--split-panel-handle-color: #{$background};
  --#{$prefix}--split-panel-handle-border-color: #{$border-interactive};
  --#{$prefix}--split-panel-handle-size: 3px;

  border-color: #{$border-interactive};
}

[class*='#{$prefix}--split-panel__handle--']:where(:focus, :hover) {
  outline: none;
}

.#{$prefix}--split-panel__handle--horizontal {
  --#{$prefix}--split-panel-handle-block-size: calc(100% - #{$spacing-03});
}

.#{$prefix}--split-panel__handle--horizontal::after {
  border-block: initial;
}

.#{$prefix}--split-panel__handle--vertical {
  --#{$prefix}--split-panel-handle-inline-size: calc(100% - #{$spacing-03});
}

.#{$prefix}--split-panel__handle--vertical::after {
  border-inline: initial;
}

.#{$prefix}--split-panel__handle--horizontal:where(:focus, :hover) {
  --#{$prefix}--split-panel-handle-cursor: col-resize;
}

.#{$prefix}--split-panel__handle--vertical:where(:focus, :hover) {
  --#{$prefix}--split-panel-handle-cursor: row-resize;
}

.#{$prefix}--split-panel__handle-icon {
  position: absolute;
  inset-block-start: 50%;
  inset-inline-start: 50%;
  transform: translate(-50%, -50%);
}
