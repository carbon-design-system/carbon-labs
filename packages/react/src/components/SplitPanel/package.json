{
  "name": "@carbon-labs/react-split-panel",
  "version": "0.4.0",
  "publishConfig": {
    "access": "public",
    "provenance": true
  },
  "type": "module",
  "description": "Carbon Labs - Split Panel",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/carbon-design-system/carbon-labs",
    "directory": "src/components/SplitPanel"
  },
  "types": "./es/index.d.ts",
  "exports": {
    ".": {
      "import": "./es/index.js",
      "require": "./lib/index.js"
    },
    "./es/*": "./es/*",
    "./lib/*": "./lib/*",
    "./scss/*": "./scss/*"
  },
  "files": [
    "es",
    "lib",
    "scss"
  ],
  "scripts": {
    "build": "node ../../../tasks/build.js",
    "clean": "rm -rf {es,lib,scss}"
  },
  "devDependencies": {
    "@carbon-labs/utilities": "canary"
  }
}
