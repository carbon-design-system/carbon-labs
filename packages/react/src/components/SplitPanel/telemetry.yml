# yaml-language-server: $schema=https://unpkg.com/@ibm/telemetry-config-schema@v1/dist/config.schema.json

version: 1
projectId: c0e6b5ea-e44d-4fbc-a40a-b5f18bd8ed93
endpoint: https://www-api.ibm.com/ibm-telemetry/v1/metrics
collect:
  jsx:
    elements:
      allowedAttributeNames:
        # SplitPanel
        - childrenAfterSplit
        - childrenBeforeSplit
        - className
        - defaultSplitValue
        - gap
        - onChange
        - orientation
        - splitKeyStepPixels
        - splitMax
        - splitMin
        - splitShiftKeyStepPixels
        # React
        - key
        - ref
      allowedAttributeStringValues:
        # SplitPanel - gap
        - default
        - narrow
        - none
        # SplitPanel - orientation
        - horizontal
        - vertical
  npm:
    dependencies: null
  js:
    functions: {}
    tokens: null
