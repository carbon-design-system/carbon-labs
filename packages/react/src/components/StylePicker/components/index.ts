/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export { StylePickerColor } from './style-picker-color';
export { StylePickerIcon } from './style-picker-icon';
export { StylePickerGroup } from './style-picker-group';
export { StylePickerSection } from './style-picker-section';
export { StylePickerSections } from './style-picker-sections';
export { StylePickerOption } from './style-picker-option';
export { StylePicker } from './style-picker';
