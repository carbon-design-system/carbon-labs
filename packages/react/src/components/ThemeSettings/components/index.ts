/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export { ThemeSettings } from './ThemeSettings';
export { ThemeSwitcher } from './ThemeSwitcher';
export { ThemeSetDropdown } from './ThemeSetDropdown';
export { ThemeMenuComplement } from './ThemeMenuComplement';
export { themeSets } from './theme-settings-consts';
