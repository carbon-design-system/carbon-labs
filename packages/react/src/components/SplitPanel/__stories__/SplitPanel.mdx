import { ArgTypes, Canvas, Meta } from '@storybook/blocks';
import * as SplitPanelStories from './SplitPanel.stories';

<Meta isTemplate />

# SplitPanel

- **Initiative owner(s):** Lee Chase
- **Status:** Draft
- **Target library:** TBD
- **Target library maintainer(s) / PR Reviewer(s):** N/A
- **Support channel:** `#carbon-labs`

{/* <!-- START doctoc generated TOC please keep comment here to allow auto update --> */}
{/* <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE --> */}

> ðŸ’¡ Check our
> [Stackblitz](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/SplitPanel)
> example implementation.

[![Edit carbon-labs](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/SplitPanel)

## Table of Contents

- [Overview](#overview)
- [Getting started](#getting-started)

{/* <!-- END doctoc generated TOC please keep comment here to allow auto update --> */}

## Overview

This component is used to split an area horizontally or vertically. It renders
two panels `childrenBeforeSplit` and `childrenAfterSplit` with a split handle in
between.

The split handle can be moved using the mouse or keyboard.

### Mouse interaction

The split handle can be dragged between the minimum to maximum values.

### keyboard interaction

When the split handle has focus it can be moved using the cursor keys. A
secondary mode, typically used for moving at a different speed can be activated
by pressing shift at the same time. The default movement sizes are 5px without
shift and 25px with shift.

The Home/End keys move the split to the splitMin/splitMax values respectively.
The default splitMin is 0, splitMax 100.

### Events

The onChange event reports the current split value. It may be wise to monitor
this event and disable interactive controls when a panel is too small to be
practical.

<Canvas of={SplitPanelStories.Default} />

## Getting started

Here's a quick example to get you started.

```bash
yarn add @carbon/react
yarn add @carbon-labs/react-split-panel
```

### JS (via import)

```javascript
import { SplitPanel } from '@carbon-labs/react-split-panel/es/index';

function App() {
  return (
    <SplitPanel
      childrenBeforeSplit={
        <div>
          <h2>Children before split</h2>
          <ul>
            <li>One</li>
            <li>Two</li>
            <li>Three</li>
          </ul>
        </div>
      }
      childrenAfterSplit={
        <div>
          <h2>Children after split</h2>
          <ul>
            <li>Apple</li>
            <li>Banana</li>
            <li>Pear</li>
          </ul>
        </div>
      }
      {...args}
    />
  );
}

export default App;
```

### SCSS

In your styles file import

````
@use '@carbon/react';
@use '@carbon-labs/react-split-panel/scss/split-panel';
```./SplitPanel.stories
````

## Panel backgrounds and layers

Like the tile component the panels of the SplitPanel component are set to the
current layer color. You should not modify this directly, doing so may affect
the visibility of the split handle. Where you are using the SplitPanel component
inside a layer, simply wrap with a layer component.

## Component API

<ArgTypes />
