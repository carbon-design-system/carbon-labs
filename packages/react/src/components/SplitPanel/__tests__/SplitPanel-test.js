/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';

import { SplitPanel } from '../components/SplitPanel';
jest.mock('./split-panel.scss', () => ({}));
describe('SplitPanel', () => {
  describe('renders as expected - Component API', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <SplitPanel
          childrenBeforeSplit={<div>before</div>}
          childrenAfterSplit={<div>after</div>}
        />
      );
      expect(container).toMatchSnapshot();
    });

    it('renders a expected parts', () => {
      render(
        <SplitPanel
          childrenBeforeSplit={<div>before</div>}
          childrenAfterSplit={<div>after</div>}
        />
      );

      screen.getByText('before');
      screen.getByText('after');
      screen.getByRole('separator');
    });

    it('supports a custom class name', () => {
      const { container } = render(
        <SplitPanel
          className="test"
          childrenBeforeSplit={<div>before</div>}
          childrenAfterSplit={<div>after</div>}
        />
      );
      expect(container.firstElementChild).toHaveClass('test');
    });

    it('supports additional props', () => {
      const { container } = render(
        <SplitPanel
          data-testid="test-button"
          childrenBeforeSplit={<div>before</div>}
          childrenAfterSplit={<div>after</div>}
        />
      );
      expect(
        // eslint-disable-next-line
        container.firstElementChild
      ).toHaveAttribute('data-testid', 'test-button');
    });
  });
});
