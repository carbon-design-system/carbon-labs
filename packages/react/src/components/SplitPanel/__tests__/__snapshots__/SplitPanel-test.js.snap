// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitPanel renders as expected - Component API should match snapshot 1`] = `
<div>
  <div
    class="clabs--split-panel"
  >
    <div
      class="clabs--split-panel__content--horizontal clabs--split-panel__content--gap-default"
      style="--clabs--split-panel-before-size: 0.5fr; --clabs--split-panel-after-size: 0.5fr;"
    >
      <div
        class="clabs--split-panel__panel"
      >
        <div>
          before
        </div>
      </div>
      <div
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="clabs--split-panel__handle--horizontal"
        data-key-shift-step="25"
        data-key-step="5"
        role="separator"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="clabs--split-panel__handle-icon"
          fill="currentColor"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4L12 15 5.83 15 8.41 12.41 7 11 2 16 7 21 8.41 19.59 5.83 17 12 17 12 28 14 28 14 4 12 4zM25 11L23.59 12.41 26.17 15 20 15 20 4 18 4 18 28 20 28 20 17 26.17 17 23.59 19.59 25 21 30 16 25 11z"
          />
        </svg>
      </div>
      <div
        class="clabs--split-panel__panel"
      >
        <div>
          after
        </div>
      </div>
    </div>
  </div>
</div>
`;
